### ISSUE TITLE
Liquidsoap did not play the correct mp3 as listed in the upcoming song queue

### ISSUE THREAD

**Issue by pixel-paul** [AzuraCast/AzuraCast#4381]
**Using Docker installation method?**
Yes

**AzuraCast version:**
Rolling Release #133a943 (2021-06-16 19:34) • Docker • PHP 8.0 

**Host Operating System:**
CentOS7

**Describe the bug:**
Liquidsoap did not play the correct mp3.

I created a playlist to play one 2 hour mp3 at 10pm. The time between now and then is covered by a shuffled general rotation playlist. I needed to align to the hour (they're generally either 1 hour or 2 hours in length), so I pressed skip on the hour, the trail that is meant to play in between each show played ok, however instead of playing the mp3 that was listed to play next in the upcoming song queue, it played a random mp3 from a playlist that is only meant to play after 6pm.

Here is the outpit from liquidsoap (I have **<-- Added Comments **):

```
2021/07/05 15:40:42 [decoder:3] Method "FFMPEG" accepted "/var/azuracast/stations/ifm/media/Monday/Changing Faces (Mondays 10pm-12am)/Rotation/cf146.mp3".    **<-- This is the mp3 that was playing and then skipped on the hour**
2021/07/05 15:40:42 [ffmpeg:3] [mp3 @ 0x7fafc801f440] Estimating duration from bitrate, this may be inaccurate
2021/07/05 16:00:10 [decoder:3] Method "FFMPEG" accepted "/var/azuracast/stations/ifm/media/Trails/New SIR 2011 long splash (le hop).mp3".  **<-- This played fine and is set to play after every mp3**
2021/07/05 16:00:10 [ffmpeg:3] [mp3 @ 0x7fafcc00f180] Estimating duration from bitrate, this may be inaccurate
2021/07/05 16:00:10 [server:3] Client 172.18.0.5 disconnected.
2021/07/05 16:00:10 [server:3] New client: 172.18.0.5.
2021/07/05 16:00:10 [server:3] Client 172.18.0.5 disconnected.
2021/07/05 16:00:10 [autodj_fallback:3] Switch to rotate_39035 with forgetful transition.
2021/07/05 16:00:10 [ffmpeg:3] [mp3 @ 0x7fafc02ef380] Estimating duration from bitrate, this may be inaccurate
2021/07/05 16:00:10 [ffmpeg:3] [mp3 @ 0x7fafc02ef380] Estimating duration from bitrate, this may be inaccurate
2021/07/05 16:00:10 [requests:3] Prepared "/var/azuracast/stations/ifm/media/Trails/New SIR 2011 long splash (le hop).mp3" (RID 2).
2021/07/05 16:00:10 [requests_fallback:3] Switch to cue_requests with forgetful transition.
2021/07/05 16:00:10 [lang:3] AzuraCast Raw Response: annotate:title="Changing Faces #142",artist="A.M.B",duration="7196.87",song_id="bb42529c3a02c78da3e932d1ef257eaf",media_id="752",liq_amplify="0.00dB",playlist_id="26":/var/azuracast/stations/ifm/media/Monday/Changing Faces (Mondays 10pm-12am)/Rotation/cf142.mp3
2021/07/05 16:00:10 [decoder:3] Method "FFMPEG" accepted "/var/azuracast/stations/ifm/media/Monday/Changing Faces (Mondays 10pm-12am)/Rotation/cf142.mp3".   **<-- This didn't play**
2021/07/05 16:00:10 [cross_39071:3] Analysis: -29.571523dB / -infdB (5.96s / 5.96s)
2021/07/05 16:00:10 [crossfade:3] Simple transition: crossed, fade-in, fade-out.
2021/07/05 16:00:11 [ffmpeg:3] [mp3 @ 0x7fafcc00f180] Estimating duration from bitrate, this may be inaccurate
2021/07/05 16:00:11 [lang:3] AzuraCast Feedback Response: OK
2021/07/05 16:00:26 [requests_fallback:3] Switch to autodj_fallback with forgetful transition.
2021/07/05 16:00:26 [ffmpeg:3] [mp3float @ 0x7fafc107b8c0] Could not update timestamps for skipped samples.
2021/07/05 16:00:26 [decoder:3] Method "FFMPEG" accepted "/var/azuracast/stations/ifm/media/Friday/Vinyl Migration (4th Friday 9-10pm)/Rotation/VinylMigration#10_May28th.mp3". **<-- This is not the mp3 that is playing**
2021/07/05 16:00:26 [cross_39071:3] Analysis: -57.443843dB / -11.504528dB (5.99s / 6.01s)
2021/07/05 16:00:26 [crossfade:3] Simple transition: crossed, fade-in, fade-out.
2021/07/05 16:00:26 [lang:3] AzuraCast Feedback Response: OK
```

**Steps to reproduce:**
Hard to be specific, however I created a playlist and needed to skip on the hour to start the playlists correctly. I did run the 'System Debugger' to recreate the song queue after I had made some changes

**Expected behavior:**
The playback to follow the upcoming song queue

**Comment by pixel-paul** [comment]
The next song that has played is the song I would have expected to play:

2021/07/05 17:01:09 [ffmpeg:3] [mp3float @ 0x7fafc107b8c0] Could not update timestamps for discarded samples.

2021/07/05 17:01:10 [playlist_friday_4:2] Finished with a non-existent file?! Something may have been moved or destroyed during decoding. It is VERY dangerous, avoid it!

2021/07/05 17:01:10 [ffmpeg:3] [mp3 @ 0x7fafc0290fc0] Estimating duration from bitrate, this may be inaccurate
2021/07/05 17:01:10 [ffmpeg:3] [mp3 @ 0x7fafc0290fc0] Estimating duration from bitrate, this may be inaccurate
2021/07/05 17:01:10 [playlist_daytime_rotation:3] Prepared "/var/azuracast/stations/invader_fm/media/Sunday/Freefall (Sunday 10pm-midnight)/Rotation/freefall_933.mp3" (RID 62).  **<-- this didn't play**
2021/07/05 17:01:10 [schedule_switch:3] Switch to cue_playlist_daytime_rotation with forgetful transition.
2021/07/05 17:01:10 [rotate_39035:3] Switch to map_metadata_39029 with forgetful transition.
2021/07/05 17:01:10 [ffmpeg:3] [mp3 @ 0x7fafc10e8a80] Estimating duration from bitrate, this may be inaccurate
2021/07/05 17:01:10 [ffmpeg:3] [mp3 @ 0x7fafc0172440] Estimating duration from bitrate, this may be inaccurate
2021/07/05 17:01:10 [next_song:3] Prepared "/var/azuracast/stations/invader_fm/media/Monday/Changing Faces (Mondays 10pm-12am)/Rotation/cf142.mp3" (RID 4).
2021/07/05 17:01:10 [autodj_fallback:3] Switch to dynamic_startup with forgetful transition.
2021/07/05 17:01:10 [dynamic_startup:3] Switch to cue_next_song.
2021/07/05 17:01:10 [cross_39071:3] Analysis: -33.846894dB / -14.275534dB (6.00s / 6.00s)
2021/07/05 17:01:10 [crossfade:3] Simple transition: crossed, fade-in, fade-out.
2021/07/05 17:01:10 [lang:3] AzuraCast Feedback Response: OK

**Comment by Abqehebter** [comment]
Do you have "Avoid duplicate artists/titles" turned on? I assume that the AutoDJ plays only what it can play in the order. If the next artist is the same one that was playing 7 minutes ago, then it's clear that he has to choose a different song there.

**Comment by pixel-paul** [comment]
Yes, that is checked, however that doesn't explain why it played a song from an explicitly set night time playlist. I'd expect that the playback queue has already worked out what has and hasn't played so will be correct in terms of avoiding duplicate artists / tracks.

The issue here appears to be the playback of a track from the Friday#4 playlist:

2021/07/05 17:01:10 [playlist_friday_4:2] Finished with a non-existent file?! Something may have been moved or destroyed during decoding. It is VERY dangerous, avoid it!

Firstly why did it play this track and not the one it said it was going to, which was VinylMigration#10_May28th.mp3:

2021/07/05 16:00:26 [decoder:3] Method "FFMPEG" accepted "/var/azuracast/stations/ifm/media/Friday/Vinyl Migration (4th Friday 9-10pm)/Rotation/VinylMigration#10_May28th.mp3". **<-- This is not the mp3 that is playing**

Both of the tracks are in the Friday#4 playlist, so that could be a bug.

Secondly why did the Friday#4 playlist even get played as it was explicitly set to only play on the 25/06/21 at 8pm:

![Screenshot 2021-07-06 at 09-51-09 Playlists](https://user-images.githubusercontent.com/389303/124571551-cb4c1900-de3f-11eb-9fd8-045a6da4b97c.png)

**Comment by Vaalyn** [comment]
This is a know issue with Liquidsoaps `request.dynamic.list` which is used in AzuraCast to fetch the next song to play.

The gist is that this method from LS basically instantly falls back to the generated .pls files and picking something from that instead of waiting for the request to return when something like a `skip` happens.

This issue is planned to be fixed with Liquidsoap 2.0 and I have already started to work on updating AzuraCast for compatibility with the Liquidsoap `2.0.0-beta2` version.

Refer to #3821 for more information.

Closing this as duplicate.
